SECURITY.md

```diff
@@ -2,7 +2,7 @@
 
 ## Supported Versions
 
-Qiskit (and `qiskit-terra`) supports one minor version release at a time, both for bug and
+Qiskit supports one minor version release at a time, both for bug and
```

Comment:
We actually have to update this for the new support window in the 1.0 era, but for the purposes of this PR it's enough to just remove references to Terra, thanks.